src/D2L.CodeStyle.Analyzers/ApiUsage/JsonParamBinderAttribute/JsonParamBinderAnalyzerFixer.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+
+namespace D2L.CodeStyle.Analyzers.ApiUsage.JsonParamBinderAttribute {
+
+	[ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(JsonParamBinderAnalyzerFixer) )]
+	public class JsonParamBinderAnalyzerFixer : CodeFixProvider {
+
+		public override FixAllProvider GetFixAllProvider() {
+			return WellKnownFixAllProviders.BatchFixer;
+		}
+
+		public override async Task RegisterCodeFixesAsync( CodeFixContext context ) {
+			Diagnostic diagnostic = context.Diagnostics.First();
+			TextSpan sourceSpan = diagnostic.Location.SourceSpan;
+
+			SyntaxNode root = await context.Document.GetSyntaxRootAsync(context.CancellationToken);
+
+			AttributeSyntax oldAttribute = root.FindToken(sourceSpan.Start)
+				.Parent.AncestorsAndSelf().OfType<AttributeSyntax>().First();
+
+			context.RegisterCodeFix( 
+				CodeAction.Create( 
+					Diagnostics.ObsoleteJsonParamBinder.Title.ToString(),
+					ct => SwapAttributes( context.Document, oldAttribute, ct)
+				), 
+				diagnostic
+			);
+		}
+
+		private async Task<Document> SwapAttributes( Document document, AttributeSyntax oldAttribute, CancellationToken ct ) {
+			SyntaxNode root = await document.GetSyntaxRootAsync(ct);
+
+			AttributeSyntax newAttribute = SyntaxFactory.Attribute(
+				SyntaxFactory.IdentifierName( "JsonConvertParameterBinder" ) );
+
+			SyntaxNode newRoot = root.ReplaceNode(oldAttribute, newAttribute );

```

Comment:
Relying on a second (the standard one) fixit for adding the using if necessary?
Minorly annoying I guess but that's fine by me. I have a thing to help do that in another repo but maybe it's better to just keep this simple.

Reply:
Yeah, I mean we could bring it out and I did look at this. However for the diagnostic to register you would have had to had the using already. Didn't want to complicate it too much

Reply:
Oh, are they from the same namespace?

Reply:
Amazing, they are.